pkg/app/piped/cloudprovider/lambda/function_test.go

```diff
@@ -174,3 +176,197 @@ func TestparseFunctionManifest(t *testing.T) {
+    path: hello-world
+  handler: app.lambdaHandler
+  runtime: nodejs14.x
+  memory: 512
+  timeout: 30
+`),
+			expected: []*model.ArtifactVersion{
+				{
+					Kind:    model.ArtifactVersion_GIT_SOURCE,
+					Version: "dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603",
+					Name:    "username/lambda-function-code",
+					Url:     "https://github.com/username/lambda-function-code/commit/dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603",
+				},
+			},
+			expectedErr: false,
+		},
+		{
+			name: "[From Source Code] ok: using source code from gitlab",
+			input: []byte(`
+apiVersion: pipecd.dev/v1beta1
+kind: LambdaFunction
+spec:
+  name: SimpleSourceCodeFunction
+  role: arn:aws:iam::76xxxxxxx:role/lambda-role
+  source:
+    git: git@gitlab.com:username/lambda-function-code.git
+    ref: dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603
+    path: hello-world
+  handler: app.lambdaHandler
+  runtime: nodejs14.x
+  memory: 512
+  timeout: 30
+`),
+			expected: []*model.ArtifactVersion{
+				{
+					Kind:    model.ArtifactVersion_GIT_SOURCE,
+					Version: "dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603",
+					Name:    "username/lambda-function-code",
+					Url:     "https://gitlab.com/username/lambda-function-code/commit/dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603",
+				},
+			},
+			expectedErr: false,
+		},
+		{
+			name: "[From Source Code] ok: using source code from bitbucket",
+			input: []byte(`
+apiVersion: pipecd.dev/v1beta1
+kind: LambdaFunction
+spec:
+  name: SimpleSourceCodeFunction
+  role: arn:aws:iam::76xxxxxxx:role/lambda-role
+  source:
+    git: git@bitbucket.org:username/lambda-function-code.git
+    ref: dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603
+    path: hello-world
+  handler: app.lambdaHandler
+  runtime: nodejs14.x
+  memory: 512
+  timeout: 30
+`),
+			expected: []*model.ArtifactVersion{
+				{
+					Kind:    model.ArtifactVersion_GIT_SOURCE,
+					Version: "dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603",
+					Name:    "username/lambda-function-code",
+					Url:     "https://bitbucket.org/username/lambda-function-code/commits/dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603",
+				},
+			},
+			expectedErr: false,
+		},
+		{
+			name: "[From Source Code] ok: using source code from other git provider",
+			input: []byte(`
+apiVersion: pipecd.dev/v1beta1
+kind: LambdaFunction
+spec:
+  name: SimpleSourceCodeFunction
+  role: arn:aws:iam::76xxxxxxx:role/lambda-role
+  source:
+    git: git@ghe.github.com:username/lambda-function-code.git
+    ref: dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603
+    path: hello-world
+  handler: app.lambdaHandler
+  runtime: nodejs14.x
+  memory: 512
+  timeout: 30
+`),
+			expected: []*model.ArtifactVersion{
+				{
+					Kind:    model.ArtifactVersion_GIT_SOURCE,
+					Version: "dede7cdea5bbd3fdbcc4674bfcd2b2f9e0579603",
+					Name:    "username/lambda-function-code",
+					Url:     "git@ghe.github.com:username/lambda-function-code.git",
+				},
+			},
+			expectedErr: false,
+		},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
```

Comment:
looks like we don't need this, since the tests do not run in parallel at test case level

Reply:
Sorry I want to run in parallel at test case level, but forgot to insert `t.Parallel()` inside `t.Run`.
I'll fix it🙏